### PR TITLE
⬆️ Upgrade to react 18, latest @testing-library and migrate to vitest

### PR DESCRIPTION
**Pull request by TheHaff**
Thought I'd open a pr with what I had so far, in case anyone knows how to fix the last issue. 

- Bump to React 18
- Migrate away from deprecated testing-library/react-hooks

TODO:
- Had to skip one test in useIsClient since i couldn't figure out ssr and hydration yet.

### CODE REVIEW

File: src/useIsClient/useIsClient.test.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { renderHook } from '@testing-library/react'
+
+import useIsClient from './useIsClient'
+
+// TODO: currently don't know how to simulate hydration of hooks.
+describe('useIsClient()', () => {
+  it.skip('should be false when rendering on the server', async () => {
```
Comment (TheHaff) [inline src/useIsClient/useIsClient.test.tsx:7]: Any help here would be appriciated

File: src/useCopyToClipboard/useCopyToClipboard.test.ts [hunk 2]
```diff
@@ -16,8 +16,6 @@ describe('useClipboard()', () => {
 
   afterEach(() => {
     jest.resetAllMocks()
-    // @ts-ignore mock clipboard
-    global.navigator.clipboard = originalClipboard
```
Comment (juliencrn) [inline src/useCopyToClipboard/useCopyToClipboard.test.ts:20]: Why?

File: src/useIsClient/useIsClient.test.tsx (hunk 1 above)
Comment (juliencrn) [reply src/useIsClient/useIsClient.test.tsx:7]: For now, we haven't any clear direction about that

https://github.com/testing-library/react-testing-library/issues/1120

**Review by juliencrn — CHANGES_REQUESTED** [review]
Hi, nice add, we will merge it when we'll be able to test both CSR and SSR. Thanks!

**Comment by juliencrn** [comment]
usehooks-ts changed since your PR, I rebased it to the latest master commit but I can't push it on your repo

File: src/useCopyToClipboard/useCopyToClipboard.test.ts (hunk 2 above)
Comment (TheHaff) [reply src/useCopyToClipboard/useCopyToClipboard.test.ts]: i guess i messed up merging, fixed now

**Comment by juliencrn** [comment]
I will merge it finally, nothing happened for `useIsClient`, we will do it later in another PR. 
Thank you @TheHaff 👍

**Pull request merged**